sdk/servicebus/perf-tests/service-bus/test/receiveBatch.spec.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { ServiceBusReceiver, ServiceBusSender } from "@azure/service-bus";
+import { PerfOptionDictionary } from "@azure/test-utils-perf";
+import { ServiceBusTest } from "./sbBase.spec";
+
+interface ReceiverOptions {
+  "max-message-count": number;
+  "number-of-messages": number;
+  "message-body-size-in-bytes": number;
+}
+
+export class BatchReceiveTest extends ServiceBusTest<ReceiverOptions> {
+  receiver: ServiceBusReceiver;
+  public options: PerfOptionDictionary<ReceiverOptions> = {
+    "number-of-messages": {
+      required: true,
+      description: "Total number of messages to send",
+      shortName: "max",
+      longName: "maximum",
+      defaultValue: 10000
+    },
+    "message-body-size-in-bytes": {
+      required: true,
+      description: "Size of each message body in bytes",
+      shortName: "size",
+      longName: "size-in-bytes",
+      defaultValue: 2000
+    },
+    "max-message-count": {
+      required: true,
+      description: "Max number of messages to receive",
+      shortName: "max",
+      longName: "maximum",
+      defaultValue: 50
+    }
+  };
+
+  constructor() {
+    super();
+    this.receiver = ServiceBusTest.sbClient.createReceiver(BatchReceiveTest.queueName, {
+      receiveMode: "receiveAndDelete"
+    });
+  }
+
+  /**
+   * sends the messages to be received later
+   */
+  public async globalSetup(): Promise<void> {
+    await super.globalSetup();
+    const sender = ServiceBusTest.sbClient.createSender(BatchReceiveTest.queueName);
+
+    const {
+      "number-of-messages": { value: numberOfMessages },
+      "message-body-size-in-bytes": { value: messageBodySize }
+    } = this.parsedOptions;
+
+    // Send messages to be able to receive as part of the test
+    await sendMessages(sender, numberOfMessages, messageBodySize);
+  }
+
+  public async runBatch(): Promise<number> {
+    const messages = await this.receiver.receiveMessages(
+      this.parsedOptions["max-message-count"].value,
+      { maxWaitTimeInMs: 500 }
+    );
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    for (const _ in messages) {
+      // empty block
+    }
```

Comment:
… uhh, why is this empty? is this needed?

Reply:
Imagining a user scenario, anyone who's receiving messages will do some processing on them, they'd at least iterate over the batch they just received.
This is to represent such a scenario.

Reply:
Can you add that comment instead? It will help other people understand that the empty block is intentional.

Perhaps you could change this loop into a comment block, since the loop currently adds unnecessary overhead to the execution of this test.

Reply:
We want the overhead to represent the scenario during execution.

Can add a comment,